pkg/io/opener.go

```diff
@@ -51,6 +52,7 @@ type (
 type Opener interface {
 	Reader(ctx context.Context, path string) (ReadCloser, error)
 	Writer(ctx context.Context, path string) (WriteCloser, error)
+	Upload(ctx context.Context, uploads map[string]UploadFunc) error
```

Comment:
Not sure if this belongs in here, it essentially just wraps the `Writer`, right? Feels to me like it should be a level above the `Opener`

Reply:
Yup that's the question I'm asking above:
* Move implementation from gcs to opener package
* Keep implementation in gcs package (replace gcsObjectWriter through something generic which uses the opener)
* Create a new package which replaces gcs (e.g. blobstorage) and leverages opener